internal/config/config.go

```diff
@@ -241,10 +307,15 @@ type Config struct {
 func (config *Config) Validate() error {
 	isValid := isChainsValid(config.Chains)
 
+	// Validate global config.
+	isValid = isValid && config.Global.Routing.isRoutingConfigValid()
+
 	if !isValid {
 		return errors.New("invalid config found")
 	}
 
+	config.Global.Routing.setDefaults()
```

Comment:
I was thinking more of a completely separate path for validating and setting defaults, something like this:

```
func (c *SingleChainConfig) setDefaults() bool {
        c.Routing.setDefaults()
}

func (c *GlobalConfig) setDefaults() bool {
        c.Routing.setDefaults()
}

func (config *Config) setDefaults() {
	config.Global.setDefaults()
        
        for chainIndex := range config.Chains {
               chainConfig := &chainsConfig[chainIndex]
               chainConfig.setDefaults()
        }
}

func parseConfig(configBytes []byte) (Config, error) {
	config := Config{}

	err := yaml.Unmarshal(configBytes, &config)
	if err != nil {
		return config, err
	}

        config.setDefaults()

	err = config.Validate()

	return config, err
}
```

Validating the config as the final step ensures that we have a completely valid config before passing it off to somewhere else.